src/vs/base/parts/storage/common/storage.ts

```diff
@@ -88,11 +93,23 @@ export enum StorageState {
 	Closed
 }
 
+export enum StorageChangeSource {
```

Comment:
Sorry, going back to this. As a client of storage service it is probably very hard to understand how to deal with the `source` properly. I wonder if we should make this more explicit and simply have a `boolean` on the event to signal that the storage changed because of a larger storage change operation such as profiles or roaming and not from a simple `store` call. I think we only want to give clients a way to tell apart the two, so maybe that would work better?

Reply:
Or we keep the enum but really find good names and good documentation...

Reply:
:+1: I had a really hard time finding a decent name here, boolean is easier